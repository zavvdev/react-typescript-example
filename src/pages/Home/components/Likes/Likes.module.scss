@import 'src/styles/theme';

.btn {
  border: 1px solid $cGrayLight;
  background: none;
  cursor: pointer;
  padding: 10px;
  font-size: 20px;
  border-radius: 5px;
  margin-left: 10px;
  user-select: none;

  span {
    margin-left: 7px;
    color: $cBlackLight;
  }
}