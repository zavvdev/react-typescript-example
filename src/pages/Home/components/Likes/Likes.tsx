import { useTranslation } from 'react-i18next';
import Namespace from 'src/i18n/namespace';
import { useAppDispatch, useAppSelector } from 'src/store/hooks';
import { selectLikesAmount } from 'src/store/likes/selectors';
import { likesActions } from 'src/store/likes/slice';
import classes from 'src/pages/Home/components/Likes/Likes.module.scss';

const Likes = () => {
  const dispatch = useAppDispatch();
  const likesAmount = useAppSelector(selectLikesAmount);
  const { t } = useTranslation(Namespace.Home);

  const handleLike = () => {
    dispatch(likesActions.increment());
  };

  return (
    <button
      type="button"
      onClick={handleLike}
      className={classes.btn}
    >
      {t('buttons.likes')}
      <span>{likesAmount}</span>
    </button>
  );
};

export default Likes;
