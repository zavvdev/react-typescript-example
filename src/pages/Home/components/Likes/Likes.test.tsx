import { screen, fireEvent } from '@testing-library/react';
import render from 'src/tests/utils/render';
import Likes from 'src/pages/Home/components/Likes/Likes';

describe('<Likes />', () => {
  test('should render Likes component (snapshot)', () => {
    const { asFragment } = render(<Likes />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should increase likes amount', () => {
    render(<Likes />);
    const btn = screen.getByText('buttons.likes');
    expect(screen.getByText('1')).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByText('1')).not.toBeInTheDocument();
    expect(screen.getByText('2')).toBeInTheDocument();
  });
});
