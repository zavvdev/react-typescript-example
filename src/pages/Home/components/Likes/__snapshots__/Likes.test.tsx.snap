// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Likes /> should render Likes component (snapshot) 1`] = `
<DocumentFragment>
  <button
    class="btn"
    type="button"
  >
    buttons.likes
    <span>
      1
    </span>
  </button>
</DocumentFragment>
`;
